test/test_system.jl

```diff
@@ -306,6 +306,11 @@ end
     sys2 = deepcopy(sys)
     @test sys2.data.time_series_storage isa IS.InMemoryTimeSeriesStorage
     @test IS.compare_values(sys, sys2)
+    # Ensure that the storage references got updated correctly.
+    for component in get_components(Component, sys2, x -> has_time_series(x))
```

Comment:
These are some extra checks I've had sitting in my sandbox. Nothing is broken or has changed. They are just some extra checks.